lib/set.js

```diff
@@ -59,13 +67,12 @@ Set.prototype.addMigration = function(title, up, down){
  */
 
 Set.prototype.save = function(fn){
-  var self = this
-    , json = JSON.stringify(this);
-  fs.writeFile(this.path, json, function(err){
-    if (err) return fn(err);
-
+  var self = this;
+  this.state.save({
```

Comment:
I must admit that I don't remember anything about this, but past-me wrote the justification here:  https://github.com/gustavnikolaj/node-migrate/commit/0c74d2f17668c64af6e129dde35c3b250b9f98a0

So it seems that it is safe, and that it will give exactly the same functionality as the present implementation, and the same drawbacks as well. If you want to persist more information, it would require changes outside of this part of the code, to make use of it.